### PR TITLE
fix: revamped rts server architecture

### LINKED ISSUE #15645: [Task]: Clean up RTS server code base and prepare it for exposing Rest API Endpoints.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

_No response_

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description

RTS node server was being used for Availability feature. Now we will be enhancing the server to utilise client side logic on the server side side like construction of AST, Evaluations logic etc. So, since our codebase was not structured for implementing such use case, so we had to revamp the structure of the codebase.
#### Architecture

- **Routes :** To define different paths to consume apis
- **Controllers :** To capture the routing and formulate the logic to provide response to a particular request
- **Middleware :** action functions that happen to be stitched in between the routes and controllers, or can be used as a part of logic that needs to be injected in between.
- **Constants :** To keep static variables as per the usage without declaring it in other parts of the codebase
- **Sockets :** Any socket apis exists inside this folder to properly structure every socket implementations
- **Utils :** Helper functions which can be used at any stage of the application

Fixes #15645

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/rts/src/constants/socket.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const APP_ROOM_PREFIX = "app:";
```
Comment (mohanarpit) [inline app/rts/src/constants/socket.ts:1]: Minor: Can we rename the file to `socket.ts` ? Keeps the naming consistent with the rest of the code base.

Comment (AmanAgarwal041) [reply app/rts/src/constants/socket.ts:1]: done : `controllers/Socket.ts` -> `controllers/socket.ts`

**Review by mohanarpit — DISMISSED** [review]

**Comment by Aishwarya-U-R** [comment]
Adding 'Test Plan Approved' since this PR is a revamp & regression run is suffice as per discussion with @AmanAgarwal041 too.

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**